diff -Naur linux-5.10.41-untouched/drivers/mfd/lpc_ich.c linux-5.10.41/drivers/mfd/lpc_ich.c
--- linux-5.10.41-untouched/drivers/mfd/lpc_ich.c	2021-05-31 12:31:54.426936965 -0500
+++ linux-5.10.41/drivers/mfd/lpc_ich.c	2021-05-31 14:26:24.659129454 -0500
@@ -489,6 +489,7 @@
 	[LPC_DH89XXCC] = {
 		.name = "DH89xxCC",
 		.iTCO_version = 2,
+		.gpio_version = PCH_DH89XXCC_GPIO,
 	},
 	[LPC_PPT] = {
 		.name = "Panther Point",
@@ -978,6 +979,7 @@
 	switch (lpc_chipset_info[priv->chipset].gpio_version) {
 	case ICH_V5_GPIO:
 	case ICH_V10CORP_GPIO:
+	case PCH_DH89XXCC_GPIO:
 		res->end = res->start + 128 - 1;
 		break;
 	default:
diff -Naur linux-5.10.41-untouched/include/linux/mfd/lpc_ich.h linux-5.10.41/include/linux/mfd/lpc_ich.h
--- linux-5.10.41-untouched/include/linux/mfd/lpc_ich.h	2021-05-31 12:31:54.862941971 -0500
+++ linux-5.10.41/include/linux/mfd/lpc_ich.h	2021-05-31 14:26:24.659129454 -0500
@@ -24,6 +24,7 @@
 	ICH_V10CORP_GPIO,
 	ICH_V10CONS_GPIO,
 	AVOTON_GPIO,
+	PCH_DH89XXCC_GPIO,
 };
 
 struct lpc_ich_info {
