diff -Naur linux-5.10.37-untouched/arch/x86/Kconfig linux-5.10.37/arch/x86/Kconfig
--- linux-5.10.37-untouched/arch/x86/Kconfig	2021-05-18 07:59:42.517336997 -0500
+++ linux-5.10.37/arch/x86/Kconfig	2021-05-18 07:45:13.912825608 -0500
@@ -2899,6 +2899,11 @@
 	def_bool y
 	depends on X86_32
 
+config MERAKI_TINKERBELL
+	bool "Meraki Tinkerbell (MX100) Support"
+	help
+	  Platform support for the meraki tinkerbell x86 platform.
+
 source "drivers/firmware/Kconfig"
 
 source "arch/x86/kvm/Kconfig"
diff -Naur linux-5.10.37-untouched/arch/x86/platform/Makefile linux-5.10.37/arch/x86/platform/Makefile
--- linux-5.10.37-untouched/arch/x86/platform/Makefile	2021-05-18 08:00:00.329391085 -0500
+++ linux-5.10.37/arch/x86/platform/Makefile	2021-05-18 07:45:13.932825785 -0500
@@ -9,6 +9,7 @@
 obj-y	+= intel/
 obj-y	+= intel-mid/
 obj-y	+= intel-quark/
+obj-y  += meraki/
 obj-y	+= olpc/
 obj-y	+= scx200/
 obj-y	+= sfi/
diff -Naur linux-5.10.37-untouched/arch/x86/platform/meraki/Makefile linux-5.10.37/arch/x86/platform/meraki/Makefile
--- linux-5.10.37-untouched/arch/x86/platform/meraki/Makefile	1969-12-31 18:00:00.000000000 -0600
+++ linux-5.10.37/arch/x86/platform/meraki/Makefile	2021-05-18 07:45:13.932825785 -0500
@@ -0,0 +1 @@
+obj-$(CONFIG_MERAKI_TINKERBELL) += tink.o
diff -Naur linux-5.10.37-untouched/arch/x86/platform/meraki/tink.c linux-5.10.37/arch/x86/platform/meraki/tink.c
--- linux-5.10.37-untouched/arch/x86/platform/meraki/tink.c	1969-12-31 18:00:00.000000000 -0600
+++ linux-5.10.37/arch/x86/platform/meraki/tink.c	2021-05-18 07:57:08.080830704 -0500
@@ -0,0 +1,181 @@
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/dmi.h>
+
+#include <linux/leds.h>
+#include <linux/leds-nu801.h>
+#include <linux/platform_device.h>
+#include <linux/gpio.h>
+#include <linux/input.h>
+#include <linux/gpio_keys.h>
+
+static struct gpio_led leds[] = {
+        {
+                .name = "mx100:green:internet", 
+                .gpio = 436,
+                .active_low = 1,
+                .default_trigger = "default-on",
+        },
+        {
+                .name = "mx100:green:lan2",
+                .gpio = 443,
+        },
+        {
+                .name = "mx100:green:lan3",
+                .gpio = 445,
+        },
+        {
+                .name = "mx100:green:lan4",
+                .gpio = 447,
+        },
+        {
+                .name = "mx100:green:lan5",
+                .gpio = 448,
+        },
+        {
+                .name = "mx100:green:lan6",
+                .gpio = 457,
+        },
+        {
+                .name = "mx100:green:lan7",
+                .gpio = 459,
+        },
+        {
+                .name = "mx100:green:lan8",
+                .gpio = 460,
+        },
+        {
+                .name = "mx100:green:lan9",
+                .gpio = 461,
+        },
+        {
+                .name = "mx100:green:lan10",
+                .gpio = 462,
+        },
+        {
+                .name = "mx100:green:lan11",
+                .gpio = 473,
+        },
+        {
+                .name = "mx100:green:ha",
+                .gpio = 441,
+                .active_low = 1,
+        },
+        {
+                .name = "mx100:orange:ha",
+                .gpio = 432,
+                .active_low = 1,
+        },
+        {
+                .name = "mx100:green:usb",
+                .gpio = 446,
+                .active_low = 1,
+        },
+        {
+                .name = "mx100:orange:usb",
+                .gpio = 444,
+                .active_low = 1,
+        },
+
+};
+
+static struct gpio_led_platform_data tink_leds_data = {
+        .num_leds = ARRAY_SIZE(leds),
+        .leds = leds,
+};
+
+static struct platform_device tink_leds_dev = {
+        .name = "leds-gpio",
+        .id = -1,
+        .dev.platform_data = &tink_leds_data,
+};
+
+static struct gpio_keys_button tink_gpio_buttons[] = {
+	{
+		.desc			= "Reset",
+		.type			= EV_KEY,
+		.code			= KEY_RESTART,
+		.gpio			= 485,
+		.active_low		= 1,
+		.debounce_interval	= 100,
+	}
+};
+
+static struct gpio_keys_platform_data tink_buttons_data = {
+	.buttons			= tink_gpio_buttons,
+	.nbuttons			= ARRAY_SIZE(tink_gpio_buttons),
+	.poll_interval			= 20,
+};
+
+static struct platform_device tink_buttons_dev = {
+	.name				= "gpio-keys-polled",
+	.id				= 1,
+	.dev = {
+		.platform_data		= &tink_buttons_data,
+	}
+};
+
+static struct led_nu801_template tink_tricolor_led_template = {
+    .device_name = "mx100",
+    .name = "tricolor",
+    .num_leds = 1,
+    .cki = 425 + 41,
+    .sdi = 425 + 6,
+    .lei = 425 + 5,
+    .ndelay = 150,
+    .init_brightness = {
+        LED_OFF,
+        LED_OFF,
+        LED_OFF,
+    },
+    .default_trigger = "none",
+    .led_colors = { "blue", "green", "red" },
+};
+
+static struct led_nu801_platform_data tink_tricolor_led_data = {
+    .num_controllers = 1,
+    .template = &tink_tricolor_led_template,
+};
+
+static struct platform_device tink_tricolor_leds = {
+    .name = "leds-nu801",
+    .id = -1,
+    .dev.platform_data = &tink_tricolor_led_data,
+};
+
+static struct platform_device tink_wdt = {
+    .name = "dh89xxx-wdt",
+};
+
+static struct platform_device *tink_devs[] __initdata = {
+        &tink_leds_dev,
+        &tink_buttons_dev,
+        &tink_tricolor_leds,
+        &tink_wdt,
+};
+
+static int __init tink_init(void)
+{
+        const char *vendor, *product;
+
+        vendor = dmi_get_system_info(DMI_SYS_VENDOR);
+        if (!vendor || strcmp(vendor, "Cisco"))
+                goto getout;
+
+        product = dmi_get_system_info(DMI_PRODUCT_NAME);
+        if (!product || strcmp(product, "MX100-HW"))
+                goto getout;
+
+        platform_add_devices(tink_devs, ARRAY_SIZE(tink_devs));
+
+        printk(KERN_INFO "Cisco Meraki MX100 Platform Setup Complete\n");
+
+
+getout:
+        return 0;
+}
+
+module_init(tink_init);
+
+MODULE_AUTHOR("Justin Delegard <jdizzle@meraki.net>");
+MODULE_DESCRIPTION("CCNG Tinkerbell Board Setup");
